### PR TITLE
Make api name editable and add expiry

### LINKED ISSUE #6302: I should be able to view the expiration date of an api key or edit the name
Currently on the show page of API keys, I can't edit the name or view  the expiration date.
Screenshot 2024-07-17 at 12 37 11

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6302

### CODE REVIEW

**Review by martmull — CHANGES_REQUESTED** [review]
Think we should use a recoilState to store the apiKey and get it in ApiKeyNameInput. Current code does not work properly as it removes the apiKeyToken display when a new api key is created

File: packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx [hunk 1]
```diff
@@ -125,6 +128,12 @@ export const SettingsDevelopersApiKeyDetail = () => {
     }
   });
 
+  useEffect(() => {
+    if (isDefined(apiKeyData)) {
+      setApiKeyName(apiKeyData.name);
+    }
+  }, [apiKeyData, setApiKeyName]);
```
[the remark this answers is not in the document]
Comment (prateekj117) [reply packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx:135]: @martmull CMIIR, but I would need to make a new atom/state for APIKey, or maybe a Familystate, as I cannot find any already for APIKey in `settings/developers/states`.

And then use those to edit the name in the main state directly inside the ApiKeyNameInput component.

[the remark this answers is not in the document]
Comment (prateekj117) [reply packages/twenty-front/src/pages/settings/developers/api-keys/SettingsDevelopersApiKeyDetail.tsx:135]: @martmull I think I can replace the use of `useEffect` event by using the onCompleted in `useFindOneRecord`.

Also, I can check the loading state and disable the input until the data is fetched.

**Comment by prateekj117** [comment]
@martmull This is ready for another round.

**Review by martmull — CHANGES_REQUESTED** [review]
added one commit to remove wrong error logging.
It does not work properly, when creating a new api Key, the token is whipped after apiKeyName is set so user cannot copy it. We need to work a little bit more on that

**Review by martmull — APPROVED** [review]
The fix required quite major changes so I did it @prateekj117. Think we are good to merge now, thank you for your contribution!

**Pull request merged**